### PR TITLE
feat: 3601 - Follow/Notes updates to My Grants

### PR DESCRIPTION
**Pull request by phm200**
## Description
When Follow/Notes feature flag is enabled, make the following updates to My Grants page:
- Rename "Shared With Your Team" tab to "Shared With My Team", no functional changes to this tab
- Remove "Interested", "Not Applying", "Applied" tabs
- Add "Followed by My Team" tab
  - List of grants followed by any user on current user's team
- Grants table column "Interested Teams" renamed to "Followed By"
- Display full name of team (not abbreviation like is shown now) for teams listed in `Followed By` and `Viewed By Columns`
- Grants table changes reflected in "Browse Grants" page as well 
- Export to CSV should reflect new column names and other changes to grants table
- URL slugs for "Shared With My Team" and "Followed by My Team" pages will reflect their titles

## Screenshots / Demo Video
**with followNotesEnabled feature flag on**
image

image

...

**with followNotesEnabled feature flag off**
image

image

### Automated and Unit Tests
- [ ] Added Unit tests

### Manual tests for Reviewer
- [ ] Added steps to test feature/functionality manually

## Checklist
- [ ] Provided ticket and description
- [ ] Provided screenshots/demo
- [ ] Provided testing information
- [ ] Provided adequate test coverage for all new code
- [ ] Added PR reviewers

### CODE REVIEW

File: packages/server/src/db/index.js [hunk 1]
```diff
@@ -822,14 +832,28 @@ async function enhanceGrantData(tenantId, data) {
...
+        .join(TABLES.users, `${TABLES.agencies}.id`, '=', `${TABLES.users}.agency_id`)
+        .join(TABLES.grant_followers, `${TABLES.users}.id`, '=', `${TABLES.grant_followers}.user_id`)
+        .whereIn('grant_id', data.map((grant) => grant.grant_id))
+        .andWhere('agencies.tenant_id', tenantId);
+    const followedBy = await followedByQuery.distinct(
+        `${TABLES.grant_followers}.grant_id`,
+        `${TABLES.grant_followers}.user_id`,
+        `${TABLES.agencies}.name as agency_name`,
+        `${TABLES.agencies}.abbreviation as agency_abbreviation`,
+    );
```
Comment (phm200) [inline packages/server/src/db/index.js:845]: should I put this query behind the feature flag as well? It won't be displayed to the user if the feature flag is off, but this is some extra work we are doing here if feature flag is off

File: packages/server/.env.example [hunk 2]
```diff
@@ -8,8 +8,15 @@ COOKIE_SECRET=itsasecretsecretsecret
 WEBSITE_DOMAIN=http://localhost:8080
 API_DOMAIN=http://localhost:8080
 
+# feature flags 
 ENABLE_GRANTS_SCRAPER=true
 SHARE_TERMINOLOGY_ENABLED=true
+ENABLE_NEW_TEAM_TERMINOLOGY=true
```
Comment (phm200) [inline packages/server/.env.example:14]: adding in the other feature flags I've seen used throughout the code for reference in the example env

File: packages/client/src/components/GrantsTable.spec.js [hunk 3]
```diff
@@ -23,6 +23,7 @@ describe('GrantsTable component', () => {
         {
           interested_agencies: [],
           viewed_by_agencies: [],
+          followed_by_agencies: [],
```
Comment (phm200) [inline packages/client/src/components/GrantsTable.spec.js:26]: this component is very lightly tested and I'd like to return to add more test coverage here, but that can be done as a follow up PR; especially as we are likely removing some code soon for feature flag removal and therefore can reduce the test surface to cover

**Comment by lsr-explore** [comment]
Looks good @phm200 !

Wonder how many teams are expected to be displayed in the table.   If it is more than say 5, it may be easier to read if they are displayed on separate lines.

**Review by lsr-explore — APPROVED** [review]

**Pull request merged**